src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -99,21 +99,29 @@ public void CompileMethod(MethodCodeNode methodCodeNodeNeedingCode)
                     CorJitFlag.CORJIT_FLG_DEBUG_INFO |
                     CorJitFlag.CORJIT_FLG_PREJIT);
 
+                CompilerTypeSystemContext typeSystemContext = _compilation.TypeSystemContext;
+
                 if (!_compilation.Options.NoLineNumbers)
                 {
-                    CompilerTypeSystemContext typeSystemContext = _compilation.TypeSystemContext;
                     IEnumerable<ILSequencePoint> ilSequencePoints = typeSystemContext.GetSequencePointsForMethod(MethodBeingCompiled);
                     if (ilSequencePoints != null)
                     {
-                        Dictionary<int, SequencePoint> sequencePoints = new Dictionary<int, SequencePoint>();
-                        foreach (var point in ilSequencePoints)
-                        {
-                            sequencePoints.Add(point.Offset, new SequencePoint() { Document = point.Document, LineNumber = point.LineNumber });
-                        }
-                        _sequencePoints = sequencePoints;
+                        SetSequencePoints(ilSequencePoints);
                     }
                 }
 
+                IEnumerable<ILLocalVariable> localVariables = typeSystemContext.GetLocalVariableNamesForMethod(MethodBeingCompiled);
```

Comment:
Right and this is certainly doable. Logically the scopes are  in the order, Func -> Block -> Vars but the current implementation assumes the order Func -> Vars.  In fact, the former is what the native compiler emits under /Od while the latter is with /O2. So, Func->Vars are not really incorrect but there are implication of debugger performance and showing all values in the function scope -- e.g, the same name of variables in different scope will appear all distinctively -- only one of them contains a real value. Coincidently the current code-gen out of RyuJit in CoreRT is always with optimization on. So the current debugging experience should be on par following the optimized mode debugging, which means not much sophisticated but at least contains a meaningful context. 

Since there are ongoing work in parallel and need more infra work in ObjectWriter, I would suggest to keep the current form (function scope) as it is, and follow this block scope up in the next step.